pkg/discovery/util/node_util.go

```diff
@@ -185,6 +192,21 @@ func DetectNodeRoles(node *api.Node) sets.String {
 	return allRoles
 }
 
+func DetectNodePools(node *api.Node) sets.String {
+	allPools := sets.NewString()
+	for k, v := range node.Labels {
```

Comment:
@pallavidn Another query. Have you come across, or you know of any case where same node would be part of more then 1 node pool.. ? It looks like  a waste to go through all labels for each node..